### ISSUE TITLE
Breakpoint problem

### ISSUE THREAD

**Issue by denis-ya** [techl/MonoRemoteDebugger#27]
### Installed product versions
- Visual Studio 2015 Professional
- This extension: compiled latest commit (f1751520b3d1d09d2e33c93c5ed4b92a06c123e3) 

### Description
I have two similar boards, breakpoints do not work with the first one, but work with the second one. Actually it throws an exception in both cases when console app finished, but for the second board I have working breakpoints.
As you can see the working (second) board has mono version 4.2.2. I tried this for the first board, but it does not work, finally I tried mono 4.4.0.
Do you have any idea how I can have working breakpoints at the first board?

[MonoRemoteDebugger.1.log.txt - does not work bp](https://github.com/techl/MonoRemoteDebugger/files/561456/MonoRemoteDebugger.1.log.txt)
[MonoRemoteDebugger.2.log.txt - work bp](https://github.com/techl/MonoRemoteDebugger/files/561457/MonoRemoteDebugger.2.log.txt)
![screen](https://cloud.githubusercontent.com/assets/17147592/19850161/169cc542-9f68-11e6-9fa7-71d6b4d9e9df.jpg)

**Comment by techcap** [comment]
Do you have firewall? The log said Network is unreachable.

**Comment by denis-ya** [comment]
No, I do not have firewall, the board connected to the laptop by Ethernet cable without any switches.
I have just found that my working board has pdb2mdb file like 
`#!/bin/sh`
`exec /usr/bin/mono --gc=sgen /usr/lib/mono/4.5/pdb2mdb.exe "$@"`
and non-working board
`exec /usr/bin/mono --gc=sgen /usr/lib/mono/4.5/pdb2mdb.exe "$@"`
I have modified this file and now breakpoints work on both boards, but still throw an exception when application finished.

**Comment by techcap** [comment]
@denis-ya So Do you mean that pdb2mdb file location was different between two boards? and so make it same path, /usr/lib/mono/4.5/pdb2mdb.exe?

**Comment by denis-ya** [comment]
No, location was the same. But non-working board had pdb2mdb without
`#!/bin/sh`
It works now, when I added this line. But still throws an exception when app completed and not all of breakpoints work. It stops on some breakpoints,  but for some breakpoints it does not stop.

**Comment by techcap** [comment]
@denis-ya Are you sure it runs only in main thread?

**Comment by denis-ya** [comment]
Sorry, what do you mean "main  thread"? Some part of code I want to debug is running from async methods. Do you mean it does not work?

**Comment by techcap** [comment]
@denis-ya Now, current version does not support breakpoints in user thread. #22

**Comment by denis-ya** [comment]
OK, thanks. So this is expected behavior, right?

**Comment by techcap** [comment]
@denis-ya yes, but it will take some time to fix it.
